src/main/java/gyro/aws/secretsmanager/SecretResource.java

```diff
@@ -0,0 +1,459 @@
+import software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.secretsmanager.model.RotationRulesType;
+import software.amazon.awssdk.services.secretsmanager.model.Tag;
+import software.amazon.awssdk.services.secretsmanager.model.TagResourceRequest;
+import software.amazon.awssdk.services.secretsmanager.model.UntagResourceRequest;
+
+/**
+ * Creates a Secret with the specified Name, Description, Kms Key, Secret String and Tags.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     aws::secret secret
+ *         name: "secret-example"
+ *         description: secret-example-description-updated"
+ *         force-delete-without-recovery: true
+ *         kms-key: $(aws::kms-key secret-kms-key-example)
+ *         secret-string: "secret-string-example"
+ *         tags: {
+ *             "secret-example-tag" : "secret-example-tag-value"
+ *         }
+ *     end
+ *
+ *     aws::kms-key secret-kms-key-example
+ *         aliases: ["alias/secret1", "alias/secret2"]
+ *         description: "secret kms key example description"
+ *     end
+ */
+@Type("secret")
+public class SecretResource extends AwsResource implements Copyable<DescribeSecretResponse> {
+
+    private String description;
+    private Boolean forceDeleteWithoutRecovery;
+    private KmsKeyResource kmsKey;
+    private Long recoveryWindowInDays;
+    private String secretBinary;
+    private String secretString;
+    private Map<String, String> tags;
+
+    // Read-only
+    private String arn;
+    private String deletedDate;
+    private String lastAccessedDate;
+    private String lastChangedDate;
+    private String lastRotatedDate;
+    private String name;
+    private String owningService;
+    private Boolean rotationEnabled;
+    private String rotationLambdaARN;
+    private RotationRulesType rotationRules;
+    private String versionId;
+    private Map<String, List<String>> versionIdsToStages;
+
+    /**
+     * The description of the secret. See `Description Info <https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_UpdateSecret.html#SecretsManager-UpdateSecret-request-Description/>`_.
+     */
+    @Updatable
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    /**
+     * The KMS master key to be used to encrypt the protected text in new versions of the secret.
+     */
+    @Updatable
+    public KmsKeyResource getKmsKey() {
+        return kmsKey;
+    }
+
+    public void setKmsKey(KmsKeyResource kmsKey) {
+        this.kmsKey = kmsKey;
+    }
+
+    /**
+     * The binary data that you want to encrypt and store in the new version of the secret. See `Secret Binary Info
+     * <https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_UpdateSecret.html#SecretsManager-UpdateSecret-request-SecretBinary/>`_.
+     */
```

Comment:
We provide doc links to user docs, when there are a lot of rules for configuring that.
```suggestion
     * The data that you want to encrypt and store in the new version of the secret. 
     */
```